Stellar-ledger-entries.x

```diff
@@ -505,15 +505,112 @@ struct ContractCodeEntry {
     opaque code<SCVAL_LIMIT>;
 };
 
+// Identifiers of all the network settings.
 enum ConfigSettingID
 {
-    CONFIG_SETTING_CONTRACT_MAX_SIZE_BYTES = 0
+    CONFIG_SETTING_CONTRACT_MAX_SIZE_BYTES = 0,
+    CONFIG_SETTING_CONTRACT_COMPUTE_V0 = 1,
+    CONFIG_SETTING_CONTRACT_LEDGER_COST_V0 = 2,
+    CONFIG_SETTING_CONTRACT_HISTORICAL_DATA_V0 = 3,
+    CONFIG_SETTING_CONTRACT_META_DATA_V0 = 4,
+    CONFIG_SETTING_CONTRACT_BANDWIDTH_V0 = 5
+};
+
+// "Compute" settings for contracts (instructions and memory).
+struct ConfigSettingContractComputeV0
+{
+    // Maximum instructions per ledger
+    int64 ledgerMaxInstructions;
```

Comment:
I wouldn't worry much about representing non-negative values via unsigned values (especially in C++). But here the main thing is that the transaction XDR typically doesn't expose uint64 parameters (IIUC for better compatibility with clients such as javascript) and these values will be compared to the transaction values. I don't feel super strongly about this, but it at least seems a bit more consistent with the existing XDR definitions.

Reply:
>transaction XDR typically doesn't expose uint64 parameters (IIUC for better compatibility with clients such as javascript)

Fwiw `int64` also exceeds the limits of JavaScript runtimes for what can be stored in a `Number` since a `Number` can only store lossless a 53-bit number and so avoiding `uint64` in favor of `int64` doesn't address the JavaScript issue. JavaScript also has `BigInt` now which alleviates the concern.

However, I've always appreciated the use of signed integers for any length or size values because it avoids underflow bugs.